client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -62,20 +65,26 @@ const PatternAssembler: Step = ( { navigation, flow, stepName } ) => {
 	const stylesheet = selectedDesign?.recipe?.stylesheet || '';
 
 	const isSidebarRevampEnabled = isEnabled( 'pattern-assembler/sidebar-revamp' );
-
-	const commonEventProps = {
-		flow,
-		step: stepName,
-		intent,
-	};
-
 	const isEnabledColorAndFonts = isEnabled( 'pattern-assembler/color-and-fonts' );
 
 	const [ selectedColorPaletteVariation, setSelectedColorPaletteVariation ] =
 		useState< GlobalStylesObject | null >( null );
 	const [ selectedFontPairingVariation, setSelectedFontPairingVariation ] =
 		useState< GlobalStylesObject | null >( null );
 
+	const recordTracksEvent = useMemo(
+		() =>
+			createRecordTracksEvent( {
+				flow,
+				step: stepName,
+				intent,
+				stylesheet,
+				color_style_slug: selectedColorPaletteVariation?.title,
+				font_style_slug: selectedFontPairingVariation?.title,
+			} ),
+		[ flow, stepName, intent ]
```

Comment:
@arthur791004 do we need to add `selectedColorPaletteVariation` and `selectedFontPairingVariation` here?

Reply:
Oh yes! Added 👍

Reply:
[nit] `color_style_slug` doesn't seem like a slug, but more of a title or name. I suggest renaming it to `color_style_title` but it's a minor detail 🙂